src/mlpack/methods/preprocess/preprocess_string_main.cpp

```diff
@@ -0,0 +1,188 @@
+  // Extracting the filename
+  const std::string filename = CLI::GetParam<std::string>("actual_dataset");
+  // This is very dangerous, Let's add a check tommorrow.
+  const std::string delimiter = CLI::GetParam<std::string>("delimeter");
+  std::vector<int> dimension = CLI::GetParam<std::vector<int>>("dimension");
+  // Sorting neccessary
+  std::sort(dimension.begin(), dimension.end());
+  // Extracting the Contents of file
+  // File pointer
+  fstream fin;
+  // Open an existing file
+  fin.open(filename, ios::in);
+  std::string line;
+  std::vector<std::string> temp;
+  std::vector<std::vector<std::string>> dataset;
+  // const size_t Column = CLI::GetParam<int>("dimension");
+  data::SplitByChar sobj(delimiter);
+  boost::string_view token, copy;
+  while (std::getline(fin, line))
+  {
+    copy = line;
+    token = sobj(copy);
+    while (!token.empty())
+    {
+      temp.push_back(std::string(token));
+      token = sobj(copy);
+    }
+    dataset.push_back(temp);
+    temp.clear();
+  }
+  fin.close();
+  // Preparing the input dataset on which string manipulation has to be done.
+  size_t row = 0;
+  std::vector<std::vector<std::string>> input(dimension.size());
+  for (size_t i = 0; i < dataset.size(); i++)
+  {
+    row = 0;
+    for (size_t j = 0; j < dataset[i].size(); j++)
+    {
+      if(row < dimension.size())
+      {
+        if (dimension[row] == j)
+        {
+          // insert into input
+          input[row].push_back(dataset[i][j]);
+          row++;          
+        }
+      }
+      else
+      {
+        break;
+      }
+    }
+  }
+  data::StringCleaning obj;
+  if (CLI::HasParam("lowercase"))
+  {
+    for (size_t i = 0; i < input.size(); i++)
+      obj.LowerCase(input[i]);
+  }
+  if (CLI::HasParam("stopwords"))
+  {
+    // Not sure how to take input for tokenizer from cli.
+    if (!CLI::HasParam("stopwordsfile"))
+    {
+      throw std::runtime_error("Please provide a file for stopwords.");
+    }
+    fstream fin;
+    // Open an existing file
+    const std::string stopwordfilename = CLI::GetParam<std::string>
+        ("stopwordsfile");
+    fin.open(stopwordfilename, ios::in);
+    std::string word;
+    std::deque<std::string> originalword;
+    std::unordered_set<boost::string_view,
+        boost::hash<boost::string_view>>stopwords;
+    while (fin>>word)
+    {
+      originalword.push_back(word);
+      stopwords.insert(originalword.back());
+    }
+    fin.close();
+    const std::string strDelimiter = CLI::GetParam<std::string>
+        ("strDelimeter");
+    for (size_t i = 0; i < input.size(); i++)
+      obj.RemoveStopWords(input[i], stopwords,
+          data::SplitByChar(strDelimiter));
+  }
+  if (CLI::HasParam("punctuation"))
+  {
+    for (size_t i = 0; i < input.size(); i++)
+      obj.RemovePunctuation(input[i]);
+  }
+  const std::string filename2 = CLI::GetParam<std::string>("preprocess"
+      "_dataset");
+  fstream fout;
+  fout.open(filename2, ios::out | ios::trunc);
```

Comment:
I suggest `ofstream`. You should verify that the file has been opened successfully.